chia/seeder/dns_server.py

```diff
@@ -395,38 +404,68 @@ async def stop(self) -> None:
-                self.pointer_v6 = 0
-                for peer in new_reliable_peers:
-                    try:
-                        validated_peer = ip_address(peer)
-                        if validated_peer.version == 4:
-                            self.reliable_peers_v4.append(validated_peer)
-                        elif validated_peer.version == 6:
-                            self.reliable_peers_v6.append(validated_peer)
-                    except ValueError:
-                        log.error(f"Invalid peer: {peer}")
-                        continue
-                log.warning(
-                    f"Number of reliable peers discovered in dns server:"
-                    f" IPv4 count - {len(self.reliable_peers_v4)}"
-                    f" IPv6 count - {len(self.reliable_peers_v6)}"
-                )
+            await self.refresh_reliable_peers()
             sleep_interval = min(15, sleep_interval + 1)
             await asyncio.sleep(sleep_interval * 60)
 
+    async def refresh_reliable_peers(self) -> None:
+        if self.crawl_store is None:
+            return
+        try:
+            new_reliable_peers = await self.crawl_store.get_good_peers()
+        except Exception as e:
+            log.error(f"Error loading reliable peers from database: {e}. Traceback: {traceback.format_exc()}.")
+            return
+        static_peers = self.config.get("static_peers", [])
+        if len(static_peers) > 0:
+            log.warning("have static peers, resolving ip addresses")
+            for static_peer in static_peers:
+                try:
+                    log.warning(f"Handling static peer {static_peer}")
+                    # Attempt to parse as an IP address
+                    # If this doesn't throw, we can just add to the list
+                    # Otherwise, we have to resolve the hostname
+                    ip_address(static_peer)
+                    new_reliable_peers.append(static_peer)
+                except ValueError:
+                    # Wasn't an IP address, so resolve the hostname
+                    log.warning(f"Not an IP address, trying to resolve {static_peer} to an IP address")
+                    if self.resolver is not None:
+                        for rdtype in ["A", "AAAA"]:
+                            result = await self.resolver.resolve(qname=static_peer, rdtype=rdtype, lifetime=30)
+                            for ip in result:
+                                try:
+                                    ip_address(ip)
+                                    new_reliable_peers.append(ip)
+                                except ValueError:
+                                    pass
+                    continue
+
+        if len(new_reliable_peers) == 0:
+            log.warning("No reliable peers found in database, waiting for db to be populated.")
+            await asyncio.sleep(2)  # sleep for 2 seconds, because the db has not been populated yet.
```

Comment:
previously, we would only sleep 2 seconds in this case. in your new behavior, we sleep 2 seconds + the 15 minutes in the outer loop. is it deliberate?

Reply:
Good catch. Put the initial group of code trying to get a list of peers in a loop which sleeps for 2s each iteration to keep that old behavior.